azkaban-common/src/test/java/azkaban/executor/ContainerizedDispatchImplTest.java

```diff
@@ -0,0 +1,266 @@
+import org.junit.Before;
+import org.junit.Test;
+
+public class ContainerizedDispatchImplTest {
+
+  private final CommonMetrics commonMetrics = new CommonMetrics(
+      new MetricsManager(new MetricRegistry()));
+  private Map<Integer, Pair<ExecutionReference, ExecutableFlow>> activeFlows = new HashMap<>();
+  private Map<Integer, Pair<ExecutionReference, ExecutableFlow>> unfinishedFlows = new
+      HashMap<>();
+  private List<Pair<ExecutionReference, ExecutableFlow>> queuedFlows = new
+      ArrayList<>();
+  private ExecutorLoader loader;
+  private ExecutorApiGateway apiGateway;
+  private ContainerizedImpl containerizedImpl;
+  private Props props;
+  private User user;
+  private ExecutableFlow flow1;
+  private ExecutableFlow flow2;
+  private ExecutableFlow flow3;
+  private ExecutableFlow flow4;
+  private ExecutionReference ref1;
+  private ExecutionReference ref2;
+  private ExecutionReference ref3;
+
+  @Before
+  public void setup() throws Exception {
+    this.props = new Props();
+    this.user = TestUtils.getTestUser();
+    this.loader = mock(ExecutorLoader.class);
+    this.apiGateway = mock(ExecutorApiGateway.class);
+    this.containerizedImpl = mock(ContainerizedImpl.class);
+    this.props.put(Constants.ConfigurationKeys.MAX_CONCURRENT_RUNS_ONEFLOW, 1);
+    this.props.put(ContainerizedExecutionManagerProperties.CONTAINERIZED_IMPL_TYPE,
+        ContainerizedImplType.KUBERNETES.name());
+    this.flow1 = TestUtils.createTestExecutableFlow("exectest1", "exec1");
+    this.flow2 = TestUtils.createTestExecutableFlow("exectest1", "exec2");
+    this.flow3 = TestUtils.createTestExecutableFlow("exectest1", "exec2");
+    this.flow4 = TestUtils.createTestExecutableFlow("exectest1", "exec2");
+    this.flow1.setExecutionId(1);
+    this.flow2.setExecutionId(2);
+    this.flow3.setExecutionId(3);
+    this.flow4.setExecutionId(4);
+    this.ref1 = new ExecutionReference(this.flow1.getExecutionId(), null);
+    this.ref2 = new ExecutionReference(this.flow2.getExecutionId(), null);
+    this.ref3 = new ExecutionReference(this.flow3.getExecutionId(), null);
+
+    this.activeFlows = ImmutableMap
+        .of(this.flow2.getExecutionId(), new Pair<>(this.ref2, this.flow2),
+            this.flow3.getExecutionId(), new Pair<>(this.ref3, this.flow3));
+    when(this.loader.fetchActiveFlows()).thenReturn(this.activeFlows);
+    this.queuedFlows = ImmutableList.of(new Pair<>(this.ref1, this.flow1));
+    when(this.loader.fetchQueuedFlows(Status.READY)).thenReturn(this.queuedFlows);
+  }
+
+  @Test
+  public void testFetchAllActiveFlows() throws Exception {
+    ContainerizedDispatchImpl containerizedDispatchImpl = initializeContainerizedDispatchImpl();
+    initializeUnfinishedFlows();
+    final List<ExecutableFlow> flows = containerizedDispatchImpl.getRunningFlows();
+    this.unfinishedFlows.values()
+        .forEach(pair -> assertThat(flows.contains(pair.getSecond())).isTrue());
+    containerizedDispatchImpl.shutdown();
```

Comment:
Please add a "@After" teardown method and add the shutdown to it.
I believe to make it happen, the containerizedDIspatchImpl must be made a member of the class.